### PR TITLE
add markdown highlighting

### LINKED ISSUE #96: Syntax highlighted markdown code (e.g. docs).
Overwrite the markdown rendered in the terminal for code blocks to write with syntax highlighting instead. Should be easy. Just use the colorscheme that is already set.

### PR DESCRIPTION
**Pull request by KristofferC**
Closes #96

### CODE REVIEW

File: src/MarkdownHighlighter.jl [hunk 1]
```diff
@@ -0,0 +1,67 @@
+
+using Crayons
+using Tokenize
+
+import OhMyREPL.Passes.SyntaxHighlighter.SYNTAX_HIGHLIGHTER_SETTINGS
+
+HIGHLIGHT_MARKDOWN = Ref(true)
+enable_highlight_markdown(v::Bool) = HIGHLIGHT_MARKDOWN[] = v
+
+function Base.Markdown.term(io::IO, md::Base.Markdown.Code, columns)
```
Comment (tkelman) [inline src/MarkdownHighlighter.jl]: isn't this going to interfere with Documenter?

Comment (KristofferC) [inline src/MarkdownHighlighter.jl]: How and why?

Comment (tkelman) [inline src/MarkdownHighlighter.jl]: by shadowing base code on base types

**Comment by KristofferC** [comment]
I guess you are saying that Documenter calls into the terminal renderer for markdown? Does it?

**Comment by tkelman** [comment]
Looks like it doesn't. The Atom or VS Code plugins might?

**Comment by KristofferC** [comment]
Atom renders Markdown using the HTML renderer and VS Code runs all their stuff in a sandboxed environment (not sure what they use).

**Comment by tkelman** [comment]
Maker.jl and Lexicon.jl appear to use `Markdown.term`

**Comment by KristofferC** [comment]
Oh, the docpackage deprecated in 0.4 uses it, hmmm. As to maker, I guess it will be slightly better when used with OMR. 

The whole point of this package is to experiment with new behaviour.if people don't love it they can disable the feature or uninstall the package.

**Comment by tkelman** [comment]
Overwriting methods like this is a bad idea, any time you find yourself doing it it's a sign that the API in question is poorly designed and really needs to be improved before 1.0 so these awful hacks can be temporary. Hiding actual problems from your users at load time to suppress legitimate warnings is a really bad tradeoff. You should not have to overwrite methods - when you do, that's a problem that needs fixing, and not by pretending it doesn't exist.

**Comment by KristofferC** [comment]
By using this package the user says that he/she rather have the OMR defaults than the defaults in Julia. You might as well consider it a super lightweight fork of Julia but Julia is so flexible that instead of me having to ship the full Julia distribution, I can just ship the diff of my "fork" vs the Base Julia and Julia will monkey patch it in for me.

>  bad idea, awful hacks, ..., Hiding actual problems, ..., suppress legitimate warnings  

You are using quite negative words about my package that I have spent a lot of time, effort and care to implement and I have tried my best to provide a good user experience and quickly fix any problems users have had. I have also been a good boy and upstreamed many of the changes that were first tried out here in this package to Base Julia, just like you are supposed to do (package -> Base).

I am tired of having these discussions every time I add a feature to this package. I know you don't like it, it has been noted but I won't be having this discussion again.

**Pull request merged**